WalletWasabi/Hwi/HwiClient.cs

```diff
@@ -262,11 +266,20 @@ public async Task<IEnumerable<HwiEnumerateEntry>> EnumerateAsync(CancellationTok
 
 		#region Helpers
 
-		private static void ThrowIfError(string responseString, IEnumerable<HwiOption> options)
+		private static void ThrowIfError(string responseString, IEnumerable<HwiOption> options, string arguments, int exitCode)
 		{
-			if (HwiParser.TryParseErrors(responseString, options, out HwiException error))
+			if (exitCode != 0)
+			{
+				if (HwiParser.TryParseErrors(responseString, options, out HwiException error))
+				{
+					throw error;
+				}
+				throw new HwiException(HwiErrorCode.UnknownError, $"'hwi {arguments}' exited with incorrect exit code: {exitCode}.");
+			}
+
+			if (HwiParser.TryParseErrors(responseString, options, out HwiException error2))
```

Comment:
Can you explain why is this trying to parse error even if `exitCode` is 0?

Reply:
It was that like before I did not change that. I think the exit code is non-zero when there is an HWI error. Exit code is Zero when HWI can successfully finish the operation but the user hardware wallet returned and error.